### ISSUE TITLE
[Phase 5] Cross-Validation Pipeline

### ISSUE THREAD

**Issue by Sakeeb91** [Sakeeb91/protein-structure-prediction#6]
## Objective

Create end-to-end pipeline with identity-aware cross-validation to prevent homolog leakage.

**Part of:** #1

## Dependencies

- All previous phases (#2-#5) complete

### Experiment Runner
- [ ] Create `src/pipeline/runner.py` with `ExperimentRunner`
- [ ] Wire together data, features, model, evaluation
- [ ] Support configuration via YAML

### Identity-Aware Splitting
- [ ] Create `src/pipeline/cv_splitter.py` with `IdentitySplitter`
- [ ] Implement sequence clustering for proper splits
- [ ] Use pre-computed clusters if available

### CLI Entry Point
- [ ] Create `scripts/run_experiment.py`
- [ ] Support dataset, model, epochs arguments

### Experiment Runner

```python
# src/pipeline/runner.py
import yaml
from src.data.loader import ProteinDataLoader
from src.features.encoding import SequenceEncoder
from src.models.cnn import ProteinCNN
from src.models.trainer import ModelTrainer
from src.evaluation.metrics import StructureMetrics

class ExperimentRunner:
    def __init__(self, config_path=None, **kwargs):
        if config_path:
            with open(config_path) as f:
                self.config = yaml.safe_load(f)
        else:
            self.config = kwargs

    def run(self):
        # Load data
        loader = ProteinDataLoader(self.config['dataset'])
        train_seqs, test_seqs, train_labels, test_labels = loader.load_split()

        # Encode
        encoder = SequenceEncoder()
        X_train = [encoder.encode(s) for s in train_seqs]
        X_test = [encoder.encode(s) for s in test_seqs]

        # Train
        model = ProteinCNN()
        trainer = ModelTrainer(model)
        for epoch in range(self.config.get('epochs', 50)):
            loss = trainer.train_epoch(train_dataloader)
            print(f"Epoch {epoch}: Loss = {loss:.4f}")

        # Evaluate
        y_pred = model.predict(X_test)
        metrics = StructureMetrics(test_labels, y_pred)

        return {
            'q3': metrics.q3(),
            'per_class': metrics.per_class_accuracy()
        }
```

### Identity-Aware Splitting Note

```python
# src/pipeline/cv_splitter.py
"""
For proper protein structure prediction evaluation,
sequences must be clustered by identity.

Homologous proteins (>25% sequence identity) should be
in the same fold to prevent data leakage.

Options:
1. Use pre-clustered datasets (CB513 is already non-redundant)
2. Use MMseqs2 for clustering: mmseqs easy-cluster seqs.fasta clusters tmp --min-seq-id 0.25
3. Use cd-hit: cd-hit -i seqs.fasta -o clusters -c 0.25
"""
```

## Definition of Done

- [ ] Full pipeline runs: `python scripts/run_experiment.py --dataset cb513`
- [ ] Results logged to JSON
- [ ] Reproducible with random seed
- [ ] Documentation on identity-based splitting

## Technical Notes

- CB513 is already non-redundant, so simple split is acceptable for initial testing
- For CullPDB, identity-based splits are critical
- Pre-defined train/test splits exist for fair comparison
